hack/bootstrap-kubestellar.sh

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: $0  --verbose
+
+# This script installs kcp and KubeStellar binaries to a folder of choice
+#
+# Arguments:
+# [--kcp-version release] set a specific kcp release version, default: latest
+# [--kubestellar-version release] set a specific KubeStellar release version, default: latest
+# [--os linux|darwin] set a specific OS type, default: autodetect
+# [--arch amd64|arm64] set a specific architecture type, default: autodetect
+# [--folder name] sets the installation folder, default: $PWD/kcp-edge
+# [--bind address] bind kcp to a specific ip address
+# [--imw name] create a Inventory Management Workspace (IMW)
+# [--wmw name] create a Workload Management Workspace (WMW)
+# [-V|--verbose] verbose output
+
+set -e
+
+kcp_installed() {
+    if [ "$(which kcp)" == "" ]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kcp_running() {
+    if [ "$(pgrep -f 'kcp start')" == "" ]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kcp_get_latest_version() {
+    curl -sL https://github.com/kcp-dev/kcp/releases/latest | grep "</h1>" | head -n 1 | sed -e 's/<[^>]*>//g' | xargs
+}
+
+kubestellar_installed() {
+    if [[ "$(which mailbox-controller)" == "" || "$(which placement-translator)" == "" || "$(which scheduler)" == "" ]]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kubestellar_running() {
+    if [[ "$(pgrep -f mailbox-controller)" == "" ||  "$(pgrep -f placement-translator)" == "" || "$(pgrep -f 'scheduler -v')" == "" ]]; then
+        echo "false"
+    else
+        echo "true"
+    fi
+}
+
+kubestellar_get_latest_version() {
+    curl -sL https://github.com/kcp-dev/edge-mc/releases/latest | grep "</h1>" | head -n 1 | sed -e 's/<[^>]*>//g' | xargs
+}
+
+get_os_type() {
+  case "$OSTYPE" in
+      linux*)   echo "linux" ;;
+      darwin*)  echo "darwin" ;;
+      *)        echo "Unsupported operating system type: $OSTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_arch_type() {
+  case "$HOSTTYPE" in
+      x86_64*)  echo "amd64" ;;
+      aarch64*) echo "arm64" ;;
+      *)        echo "Unsupported architecture type: $HOSTTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_full_path() {
+    echo "$(cd "$1"; pwd)"
+}
+
+kcp_version=""
+kubestellar_version=""
+os_type=""
+arch_type=""
+folder=""
+strip_bin="false"
```

Comment:
Nothing reads this variable. I think that is good.

Reply:
ok, removed